grommet/grommet
# [TS] Migrate SkipLinks tests to .tsx

What does this PR do?
This PR migrates MaskedInput component tests from *.js to *.tsx to allow ts-jest to catch type-related errors in the unit testing pipeline. Changes made:

1. Optional Chaining (?.) was added for document.getElementById('skip-links')?.querySelector('a') to handle potential null values.
2. Type Assertions were used in places like fireEvent.click(document.activeElement as Element); to ensure that document.activeElement is not null and is treated as an Element.
3. id prop added both in propTypes.js and index.d.ts.

Where should the reviewer start?
Ensure all tests are currently passing

What testing has been done on this PR?
Manually ran and passed MaskedInput component tests locally.
CircleCI tests have passed.

How should this be manually tested?
Modified tests behave identically as before.

Do Jest tests follow these best practices?

Any background context you want to provide?

What are the relevant issues?

Screenshots (if appropriate)

Do the grommet docs need to be updated?
no

Should this PR be mentioned in the release notes?
no

Is this change backwards compatible or is it a breaking change?
compatible
